src/confluent_kafka/src/confluent_kafka.c

```diff
@@ -816,15 +816,26 @@ static int TopicPartition_clear (TopicPartition *self) {
 		Py_DECREF(self->error);
 		self->error = NULL;
 	}
+	if (self->metadata) {
+		Py_DECREF(self->metadata);
+		self->metadata = NULL;
+	}
 	return 0;
 }
 
 static void TopicPartition_setup (TopicPartition *self, const char *topic,
 				  int partition, long long offset,
+				  const char *metadata,
```

Comment:
metadata is a binary blob, not a string, so we can't use strlen() to know the end of the data, the size must be passed explicitly. The corresponding Python type should be `bytes` (for py3, there are macros to help with py2 in confluent_kafka.h)

Reply:
This does not seem to be fixed